### PR TITLE
Move some Exif functions to apps/shared/avifexif

### PR DESCRIPTION
**Pull request by y-guyon**
So that the avif_apps target does not depend on avif_internal.

### CODE REVIEW

File: apps/shared/avifexif.c [hunk 1]
```diff
@@ -0,0 +1,63 @@
+// Copyright 2022 Google LLC. All rights reserved.
```
Comment (jzern) [inline apps/shared/avifexif.c:1]: `// Copyright 2022 Google LLC. All rights reserved.`

maybe we need a normalization pass here, I think this is now `// Copyright 2022 Google LLC`

File: apps/shared/avifexif.h [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// Copyright 2022 Google LLC. All rights reserved.
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include "avif/avif.h"
+
+// Converts image->transformFlags, image->irot and image->imir to the equivalent Exif orientation value in [1:8].
+uint8_t avifImageGetExifOrientationFromIrotImir(const avifImage * image);
+
+// Attempts to parse the image->exif payload until the Exif orientation is found, then sets it to the given value.
```
Comment (jzern) [inline apps/shared/avifexif.h:9]: This function doesn't take an avifImage*, so the comment should just say 'exif', no?

File: apps/shared/avifexif.c (hunk 1 above)
Comment (y-guyon) [reply apps/shared/avifexif.c:1]: Done in https://github.com/AOMediaCodec/libavif/pull/1201.

**Review by jzern — APPROVED** [review]

File: apps/shared/avifjpeg.c [hunk 3]
```diff
@@ -2,7 +2,7 @@
 // SPDX-License-Identifier: BSD-2-Clause
 
 #include "avifjpeg.h"
-#include "avif/internal.h"
+#include "avifexif.h"
```
Comment (wantehchang) [inline apps/shared/avifjpeg.c:5]: Nit: List headers in alphabetical order.

Make the same change to apps/shared/avifpng.c.

**Review by wantehchang — APPROVED** [review]
LGTM. Thanks.

File: apps/shared/avifexif.h [hunk 4]
```diff
@@ -0,0 +1,10 @@
+// Copyright 2022 Google LLC
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include "avif/avif.h"
+
```
Comment (wantehchang) [inline apps/shared/avifexif.h:5]: Add a header include guard and extern "C" for C++ code.

**Review by wantehchang — APPROVED** [review]
LGTM.

File: apps/shared/avifjpeg.c (hunk 3 above)
Comment (jzern) [reply apps/shared/avifjpeg.c:5]: To make that clearer we could add blank line:

https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes

**Pull request merged**